nsv13/code/modules/overmap/knpc.dm

```diff
@@ -366,11 +380,17 @@ This is to account for sec Ju-Jitsuing boarding commandos.
 			if(H.put_in_hands(target_item))
 				H.visible_message("<span class='warning'>[H] grabs [target_item]!</span>")
 
+///Checks that G exists, has ammo and that H can fire it. Returns G if yes, FALSE otherwise.
+/datum/ai_goal/human/acquire_weapon/proc/gun_suitable(mob/living/carbon/human/H, obj/item/gun/G)
+	if(!G)
+		return FALSE
+	return G.can_shoot() && G.can_trigger_gun(H) ? G : FALSE
```

Comment:
Include a check for either the KNPC having ammo for that gun or the gun itself actually having ammo.

Reply:
Whether the gun has ammo is checked in G.can_shoot()
Checking whether the knpc has ammo on themselves for the gun is a good idea, I was just too lazy to add that too. (Wasn't checked previously either and I was focussing on the firing pins. One step at a time.)

Reply:
They do sometimes get disarmed or drop weapons when climbing over things. So picking up gun if they still have mags would be a good idea.